spec/requests/user_invitations_spec.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe "User Invitations", type: :request do
+  describe "GET /users/invitation/new" do
+    context "when a system user is signed in" do
+      let(:user) { create(:user, :system) }
+      before(:each) do
+        sign_in(user)
+      end
+
+      it "renders the new template" do
+        get "/users/invitation/new"
+        expect(response).to render_template(:new)
+      end
+    end
+
+    context "when a non-system user is signed in" do
+      let(:user) { create(:user, :data_agent) }
+      before(:each) do
+        sign_in(user)
+      end
+
+      it "redirects to the permissions error page" do
+        get "/users/invitation/new"
+        expect(response).to redirect_to("/pages/permission")
+      end
+    end
+  end
+
+  describe "POST /users/invitation" do
+    let(:email) { attributes_for(:user)[:email] }
+    let(:role) { attributes_for(:user)[:role] }
+    let(:params) do
+      { user: { email: email, role: role } }
+    end
+
+    context "when a system user is signed in" do
+      let(:user) { create(:user, :system) }
+      before(:each) do
+        sign_in(user)
+      end
```

Comment:
Does `sign_in(user)` need to execute before each test or can it be executed once using `before(:context)`? If it is the latter, I suggest making this change for each context in this spec (just to avoid duplicate executions when new test cases are added).

Reply:
@eminnett I think you're right and it's probably overkill. Will make some changes!

Reply:
For the record, we tried this out but it resulted in broken tests – will keep this in mind for future specs though.

Reply:
Fine with me.